src/provider/no_op_provider.rs

```diff
@@ -1,29 +1,73 @@
+use std::sync::Arc;
+
 use async_trait::async_trait;
+use typed_builder::TypedBuilder;
 
-use crate::{EvaluationContext, StructValue};
+use crate::{
+    EvaluationContext, EvaluationContextFieldValue, EvaluationError, EvaluationReason,
+    FlagMetadata, StructValue,
+};
 
 use super::{FeatureProvider, ProviderMetadata, ResolutionDetails};
 
-const PROVIDER_NAME: &'static str = "No Operation";
+// ============================================================
+//  NoOpProvider
+// ============================================================
 
-/// The default provider that simply returns given default value during _evaluation.
-#[derive(Debug)]
+/// The default provider that does nothing.
+///
+/// By default, it returns the default value of each supported type. You can inject values (for
+/// testing purpose or simply providing some fixed value) by the following ways:
+/// 1. When creating an instance, inject values through the builder.
+/// 2. Supply a value with key "Value" in the evaluation context. Because the evaluation context is
```

Comment:
To the degree that this happens (which I don't think it should), we should probably move this to EvaluationOptions when we implement it.

Reply:
What is your perspective about this? I mean, which part should better be re-considered in your opinion? :-)

Reply:
EvaluationContext is "here is stuff I know about the user and surrounding context. Use this when making decisions"

You're doing something slightly different. You're saying "ignore all context values, and change the way your decisions are being made with this override". I'd probably push for this to be in the constructor or to use something closer to a mock. We don't really want people repeating this pattern.

Looking again, EvaluationOptions is about hooks and hook hints, so there isn't a clear fix for that. I'm fine if this is a "file a ticket for later to make a better mock provider for testing" sort of thing.

Reply:
I agree that putting it into evaluation context might bring confusions. I have modified this part that:
* The value injection will only work on the constructor of no-op provider.
* The flag metadata of no-op provider will return key-value pairs from evaluation context.

Reply:
I am going to resolve this conversation now. Please feel free to open an issue or ping me on Slack if you have to further discuss it. :-)